### PR TITLE
fix api metadata extract from context

### PR DESCRIPTION
**Pull request by vtolstov**
Signed-off-by: Vasiliy Tolstov <v.tolstov@unistack.org>

### CODE REVIEW

File: api/handler/rpc/rpc.go [hunk 1]
```diff
@@ -118,6 +118,17 @@ func (h *rpcHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
...
+	md, ok := r.Context().Value(metadata.MetadataKey{}).(metadata.Metadata)
+	if !ok {
+		md = make(metadata.Metadata)
+	}
+
+	// merge context with overwrite
+	cx = metadata.MergeContext(cx, md, true)
+
+	// set merged context to request
+	*r = *r.WithContext(cx)
```
Comment (asim) [inline api/handler/rpc/rpc.go:131]: Why dereference like this?

Comment (vtolstov) [reply api/handler/rpc/rpc.go:131]: to modify original context inside r.
r.WithContext returns pointer to http Request. So we need to modify original pointer with new (that actually have only context differs)

Comment (asim) [reply api/handler/rpc/rpc.go:131]: https://golang.org/src/net/http/request.go?s=13039:13098#L346

It already does this for you

Comment (asim) [reply api/handler/rpc/rpc.go:131]: Ah crap its making a copy

Comment (vtolstov) [reply api/handler/rpc/rpc.go:131]: no, it returns new Request pointer (r2). Mostly i need to use *r = *r.Clone(ctx)
as i saw in docs. But none of this two functions modifies original request.

**Pull request merged**